src/createServer.js

```diff
@@ -1,3 +1,69 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const supportedCases = {
+  SNAKE: true,
+  KEBAB: true,
+  CAMEL: true,
+  PASCAL: true,
+  UPPER: true,
+};
+
+const possibleErrors = {
+  text:
+    'Text to convert is required. ' +
+    'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  noCase:
+    '"toCase" query param is required. ' +
+    'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  notValidCase:
+    'This case is not supported. ' +
+    'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+};
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const { pathname, searchParams } = new URL(
+      req.url,
+      `http://${req.headers.host}`,
+    );
+    const text = pathname.slice(1);
+    const caseName = searchParams.get('toCase');
+
+    if (!text || !caseName || !supportedCases[caseName]) {
+      res.statusCode = 404;
```

Comment:
The HTTP status code for 'Bad request' is 400, not 404. You should change this to respond with the correct status code.